migrations/V186__Add_MTO_Solution.sql

```diff
@@ -53,10 +53,10 @@ COMMENT ON INDEX unique_name_per_model_plan_when_mto_common_solution_is_null IS
 
 
 ALTER TABLE mto_solution
-ADD CONSTRAINT check_name_type_and_common_solution CHECK (
-    (mto_common_solution_key IS NULL AND name IS NOT NULL AND type IS NOT NULL)
-    OR (mto_common_solution_key IS NOT NULL AND name IS NULL AND type IS NULL)
+ADD CONSTRAINT check_custom_solution_fields CHECK (
+    (mto_common_solution_key IS NULL AND ( name IS NOT NULL AND type IS NOT NULL AND poc_email IS NOT NULL AND poc_name IS NOT NULL))
+    OR (mto_common_solution_key IS NOT NULL AND ( name IS NULL AND type IS NULL AND poc_email IS NULL AND poc_name IS NULL))

```

Comment:
In doing some recent SQL work, I recently learned about the `IS DISTINCT FROM` or `IS NOT DISTINCT FROM` operator.

I wonder if this would be a different/cleaner way to write this?

```suggestion
ADD CONSTRAINT check_custom_solution_fields CHECK (
    (mto_common_solution_key IS DISTINCT FROM name)
    AND (mto_common_solution_key IS DISTINCT FROM poc_email)
    AND (mto_common_solution_key IS DISTINCT FROM poc_name)
```

By no means do we need to change it, but wanted to share my thoughts on this as I was re-reading this constraint!

Reply:
Basically a way for us to say "you can't have neither, you can't have both" with regards to these fields

Reply:
Wait -- this might not actually work the way I thought, since if they're all set to values (but those values are different) then they're still distinct...

I'll keep this around for posterity, but I think we should just ignore this suggestion 😂

Reply:
Sounds good! Yeah I think you are right that this wouldn't work as it currently stands because each area has separate values. It would be nice though!

Reply:
We might be able to simplify this with XOR-like mutual exclusion. This is untested though. Non-blocker.

```suggestion
ADD CONSTRAINT check_custom_solution_fields CHECK (
    (mto_common_solution_key IS NULL) = (name IS NOT NULL AND type IS NOT NULL AND poc_email IS NOT NULL AND poc_name IS NOT NULL)
);
```